### ISSUE TITLE
Manage of big numbers

### ISSUE THREAD

**Issue by eruizgar91** [oceanprotocol/aquarius#182]
Currently, there is an issue when big numbers are used to save data in MongoDB. We need to find the best solution could be using another type or saving as a String and managing in the different squids' implementations.

**Comment by billbsing** [comment]
This does not work when I use pleuston

**Comment by aaitor** [comment]
@billbsing , can you provide in this issue the logs, configuration of the environment used (nile, spree, etc) and some steps to reproduce please?

**Comment by billbsing** [comment]
BARGE_GIT="https://github.com/oceanprotocol/barge.git"

PLEUSTON_GIT="https://github.com/oceanprotocol/pleuston.git"
PLEUSTON_BRANCH="develop"

BRIZO_GIT="https://github.com/oceanprotocol/brizo.git"
BRIZO_TAG="v0.3.5"

AQUARIUS_GIT="https://github.com/oceanprotocol/aquarius.git"
# AQUARIUS_TAG="v0.2.3"
AQUARIUS_TAG="develop"

KEEPER_CONTRACTS_GIT="https://github.com/oceanprotocol/keeper-contracts.git"
KEEPER_CONTRACTS_TAG="v0.9.1"

OCEAN_NETWORK=nile

Run all services above, register an asset in pleuston using a price of 88.88888

**Comment by eruizgar91** [comment]
Currently the aquarius version is not update it in Nile. Can you try to reproduce using spree? If you don't find the error in Spree we will move to upgrade Nile

**Comment by billbsing** [comment]
But with Health1a, we are running the Aquarius server as a local service using the Nile network

**Comment by jernejpregelj** [comment]
Errors from my testing with latest Pleuston and Aquarius v0.2.6 in local Barge-Spree setup.
I was publishing asset with price 1400000000 tokens which converts to 1400000000000000000000000000 wei-drops-vodkas.

Publishing an assets saves it to mongo but errors on return:
```
aquarius_1                   | 2019-05-27 10:21:49 4d65ef2c5afb aquarius[10] INFO got register request: {'@context': 'https://w3id.org/did/v1', 'id': 'did:op:a657d6ea3c7e4145b45bf9fb71c72678e78a09ef60f841eb8f61fa43e8638e57', 'publicKey': [{'id': 'did:op:a657d6ea3c7e4145b45bf9fb71c72678e78a09ef60f841eb8f61fa43e8638e57', 'type': 'EthereumECDSAKey', 'owner': '0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e'}], 'authentication': [{'type': 'RsaSignatureAuthentication2018', 'publicKey': 'did:op:a657d6ea3c7e4145b45bf9fb71c72678e78a09ef60f841eb8f61fa43e8638e57'}], 'service': [{'type': 'Access', 'creator': '', 'purchaseEndpoint': 'http://localhost:8030/api/v1/brizo/services/access/initialize', 'serviceEndpoint': 'http://localhost:8030/api/v1/brizo/services/consume', 'name': 'dataAssetAccessServiceAgreement', 'templateId': '0xD306b5edCDC7819E1EB80B43De6548931706A3f4', 'serviceAgreementTemplate': {'contractName': 'EscrowAccessSecretStoreTemplate', 'events': [{'name': 'AgreementCreated', 'actorType': 'consumer', 'handler': {'moduleName': 'escrowAccessSecretStoreTemplate', 'functionName': 'fulfillLockRewardCondition', 'version': '0.1'}}], 'fulfillmentOrder': ['lockReward.fulfill', 'accessSecretStore.fulfill', 'escrowReward.fulfill'], 'conditionDependency': {'lockReward': [], 'accessSecretStore': [], 'escrowReward': ['lockReward', 'accessSecretStore']}, 'conditions': [{'name': 'lockReward', 'timelock': 0, 'timeout': 0, 'contractName': 'LockRewardCondition', 'functionName': 'fulfill', 'parameters': [{'name': '_rewardAddress', 'type': 'address', 'value': '0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e'}, {'name': '_amount', 'type': 'uint256', 'value': 1.4e+27}], 'events': [{'name': 'Fulfilled', 'actorType': 'publisher', 'handler': {'moduleName': 'lockRewardCondition', 'functionName': 'fulfillAccessSecretStoreCondition', 'version': '0.1'}}]}, {'name': 'accessSecretStore', 'timelock': 0, 'timeout': 0, 'contractName': 'AccessSecretStoreCondition', 'functionName': 'fulfill', 'parameters': [{'name': '_documentId', 'type': 'bytes32', 'value': 'a657d6ea3c7e4145b45bf9fb71c72678e78a09ef60f841eb8f61fa43e8638e57'}, {'name': '_grantee', 'type': 'address', 'value': ''}], 'events': [{'name': 'Fulfilled', 'actorType': 'publisher', 'handler': {'moduleName': 'accessSecretStore', 'functionName': 'fulfillEscrowRewardCondition', 'version': '0.1'}}, {'name': 'TimedOut', 'actorType': 'consumer', 'handler': {'moduleName': 'accessSecretStore', 'functionName': 'fulfillEscrowRewardCondition', 'version': '0.1'}}]}, {'name': 'escrowReward', 'timelock': 0, 'timeout': 0, 'contractName': 'EscrowReward', 'functionName': 'fulfill', 'parameters': [{'name': '_amount', 'type': 'uint256', 'value': 1.4e+27}, {'name': '_receiver', 'type': 'address', 'value': ''}, {'name': '_sender', 'type': 'address', 'value': ''}, {'name': '_lockCondition', 'type': 'bytes32', 'value': ''}, {'name': '_releaseCondition', 'type': 'bytes3name': '_lockCondition', 'type': 'bytes32', 'value': ''}, {'name': '_releaseCondition', 'type': 'bytes3e': ''}, {'name': '_lockCondition', 'type': 'bytes32', 'value': ''}, {'name': '_releaseCondition', 'type': 'bytes3ype': 'address', 'value': ''}, {'name': '_lockCondition', 'type': 'bytes32', 'value': ''}, {'name': '_releaseCondition', 'type': 'bytes32', 'value': ''}], 'events': [{'name': 'Fulfilled', 'actorType': 'publisher', 'handler': {'moduleName': 'escrowRewardCondition', 'functionName': 'verifyRewardTokens', 'version': '0.1'}}]}]}, 'serviceDefinitionId': '0'}, {'type': 'Authorization', 'service': 'SecretStore', 'serviceEndpoint': 'http://localhost:12001', 'serviceDefinitionId': '1'}, {'type': 'Metadata', 'serviceEndpoint': 'http://aquarius:5000/api/v1/aquarius/assets/ddo/did:op:a657d6ea3c7e4145b45bf9fb71c72678e78a09ef60f841eb8f61fa43e8638e57', 'metadata': {'curation': {'rating': 0, 'numVotes': 0, 'schema': 'Binary Voting'}, 'base': {'name': 'Title', 'description': 'Description', 'dateCreated': 'Mon May 27 2019 12:21:22 GMT+0200 (Central European Summer Time)', 'size': None, 'author': 'Author', 'type': 'dataset', 'license': 'Public Domain', 'copyrightHolder': 'Copyright', 'encoding': None, 'compression': None, 'contentType': None, 'workExample': None, 'files': [{'index': 0}], 'inLanguage': None, 'tags': 'tag1, tag2', 'price': '1400000000000000000000000000', 'encryptedFiles': '0xa03234500d620661644d7ec8899a131d52c2b0bb88cda543d3dbda921789a5ed5548bda596acfc007daa7debea897f81f40a2e7c3dfc509fc7e9fc9e1c3526', 'checksum': '66fe7c5c351a079da33c9f515aaf23998698405eab52e7a93e65dc24a5d0593a'}, 'additionalInformation': {'updateFrequency': 'seldom', 'structuredMarkup': []}}, 'serviceDefinitionId': '2'}], 'created': '2019-05-27T10:21:25Z', 'proof': {'created': '2019-05-27T10:21:27Z', 'creator': '0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e', 'type': 'DDOIntegritySignature', 'signatureValue': '0xc8a04c80a33efdd0b1cbcb7b753fe0299589d5f1c3083ceb61c1f0b9855dc21f1dda58167ddcc58b3184f10064c27f445bec023e25c1df7bd463d80e3083ed401b'}}
aquarius_1                   | 2019-05-27 10:21:49 4d65ef2c5afb aquarius[10] INFO got register request: {'name': 'Title', 'description': 'Description', 'dateCreated': 'Mon May 27 2019 12:21:22 GMT+0200 (Central European Summer T 'tags': 'tag1, tag2', 'price': '1400000000000000000000000000', 'encryptedFiles': '0xa03234500d620661644d7ec8899a131d52c2b0bb88cda543d3dbda921789a5ed5548bda596acfc007daa7debea897f81f40a2e7c3dfc509fc7e9fc9e1c3526', 'checksum': '66fe7c5c351a079da33c9f515aaf23998698405eab52e7a93e65dc24a5d0593a'}, 'additionalInformation': {'updateFrequency': 'seldom', 'structuredMarkup': []}}, 'serviceDefinitionId': '2'}], 'created': '2019-05-27T10:21:25Z', 'proof': {'created': '2019-05-27T10:21:27Z', 'creator': '0x00Bd138aBD70e2F00903268F3Db08f2D25677C9e', 'type': 'DDOIntegritySignature', 'signatureValue': '0xc8a04c80a33efdd0b1cbcb7b753fe0299589d5f1c3083ceb61c1f0b9855dc21f1dda58167ddcc58b3184f10064c27f445bec023e25c1df7bd463d80e3083ed401b'}}
aquarius_1                   | 2019-05-27 10:21:49 4d65ef2c5afb aquarius[10] INFO got register request: {'name': 'Title', 'description': 'Description', 'dateCreated': 'Mon May 27 2019 12:21:22 GMT+0200 (Central European Summer Time)', 'size': None, 'author': 'Author', 'type': 'dataset', 'license': 'Public Domain', 'copyrightHolder': 'Copyright', 'encoding': None, 'compression': None, 'contentType': None, 'workExample': None, 'files': [{'index': 0}], 'inLanguage': None, 'tags': 'tag1, tag2', 'price': '1400000000000000000000000000', 'encryptedFiles': '0xa03234500d620661644d7ec8899a131d52c2b0bb88cda543d3dbda921789a5ed5548bda596acfc007daa7debea897f81f40a2e7c3dfc509fc7e9fc9e1c3526', 'checksum': '66fe7c5c351a079da33c9f515aaf23998698405eab52e7a93e65dc24a5d0593a'}
aquarius_1                   | 2019-05-27 10:21:49 4d65ef2c5afb aquarius[10] ERROR encounterd an error while saving the asset data to OceanDB: invalid literal for int() with base 10: '1.4e+27'
aquarius_1                   | 2019-05-27 10:21:49 4d65ef2c5afb flask.app[10] ERROR Exception on /api/v1/aquarius/assets/ddo/query [POST]
aquarius_1                   | Traceback (most recent call last):
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
aquarius_1                   | Traceback (most recent call last):
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
aquarius_1                   |     response = self.full_dispatch_request()
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
aquarius_1                   |     rv = self.handle_user_exception(e)
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
aquarius_1                   |     return cors_after_request(app.make_response(f(*args, **kwargs)))
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
aquarius_1                   |     reraise(exc_type, exc_value, tb)
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
aquarius_1                   |     raise value
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
aquarius_1                   |     rv = self.dispatch_request()
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
aquarius_1                   |     return self.view_functions[rule.endpoint](**req.view_args)
aquarius_1                   |   File "/aquarius/aquarius/app/assets.py", line 643, in query_ddo
aquarius_1                   |     _sanitize_record(i)
aquarius_1                   |   File "/aquarius/aquarius/app/assets.py", line 723, in _sanitize_record
aquarius_1                   |     parameter['value'] = int(parameter['value'])
aquarius_1                   | ValueError: invalid literal for int() with base 10: '1.4e+27'
```

Making a query fails with error:
```
aquarius_1                   | 2019-05-27 10:27:14 4d65ef2c5afb flask.app[10] ERROR Exception on /api/v1/aquarius/assets/ddo/query [POST]
aquarius_1                   | Traceback (most recent call last):
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
aquarius_1                   |     response = self.full_dispatch_request()
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
aquarius_1                   |     rv = self.handle_user_exception(e)
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask_cors/extension.py", line 161, in wrapped_function
aquarius_1                   |     return cors_after_request(app.make_response(f(*args, **kwargs)))
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
aquarius_1                   |     reraise(exc_type, exc_value, tb)
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
aquarius_1                   |     raise value
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
aquarius_1                   |     rv = self.dispatch_request()
aquarius_1                   |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
aquarius_1                   |     return self.view_functions[rule.endpoint](**req.view_args)
aquarius_1                   |   File "/aquarius/aquarius/app/assets.py", line 643, in query_ddo
aquarius_1                   |     _sanitize_record(i)
aquarius_1                   |   File "/aquarius/aquarius/app/assets.py", line 723, in _sanitize_record
aquarius_1                   |     parameter['value'] = int(parameter['value'])
aquarius_1                   | ValueError: invalid literal for int() with base 10: '1.4e+27'
```

**Comment by eruizgar91** [comment]
This issue is fixed in aquarius version 0.2.8.
